### DISCUSSION TITLE
Custom GUI look?

### DISCUSSION THREAD

**Discussion by flukejones** [emilk/egui#390] [Q&A] [answered]
Not sure how to explain this beyond pointing at various embedded GUI such as printer displays, kiosks etc. I need to find a rust backed GUI crate that I can use on embedded Linux which can be fully customized (as in not window-manager style, no windows and such). egui does seem to be more heavily geared to game developers so far.

**Comment by emilk** [comment, marked as answer]
`egui` is quite customizable. You can change the colors, spacing and sizes of most everything. Check out the `Settings` window in [the web demo](https://emilk.github.io/egui/index.html).

You can also create your own custom widgets rather easily: https://github.com/emilk/egui/blob/master/egui_demo_lib/src/apps/demo/toggle_switch.rs

**Reply by felipetesc** [reply]
> `egui` is quite customizable. You can change the colors, spacing and sizes of most everything. Check out the `Settings` window in [the web demo](https://emilk.github.io/egui/index.html).
> 
> You can also create your own custom widgets rather easily: https://github.com/emilk/egui/blob/master/egui_demo_lib/src/apps/demo/toggle_switch.rs

Hey the link new_bt seems broken

**Reply by woelper** [reply]
The toggle switch has now moved to
https://github.com/emilk/egui/blob/master/crates/egui_demo_lib/src/demo/toggle_switch.rs

**Comment by follower** [comment]
Oh, hey, another Kiwi: Hi! :)

I was about to ask if you'd seen https://github.com/embedded-graphics/embedded-graphics but then realized you said "embedded *Linux*", so thought maybe your target system might be less constrained than the environments that project targets.

Are you able to expand on your "fully customized" requirement (i.e. is it just theme or also behaviour?) and what aspects seemed "more heavily geared to game developers" in a way that would make it less suitable for your scenario?

FWIW my first use of egui didn't use any of the UI widget functionality--only its ability to draw blended triangles with per-vertex colours. :D

My current project essentially keeps most of its use to one non-window area, so it's definitely possible to not appear to have windows.

P.S. Appears your web site's SSL certificate has expired. :)

**Reply by flukejones** [reply]
"fully customized" would be a fullscreen exclusive mode UI that does not use any of the visual elements the demos show, but would use the logical parts. As in I would need buttons created from images, custom scrollview and sizing, custom drawing (like canvas in flutter: https://api.flutter.dev/flutter/dart-ui/Canvas-class.html).

It looks like `DancingStrings` and the toggle switch demo show most of the above.
